examples/message_passing/after.cpp

```diff
@@ -0,0 +1,57 @@
+// This example shows how to use caf::after
+
+#include "caf/all.hpp"
+
+#include <chrono>
+#include <iostream>
+#include <random>
+
+using std::cout;
+using std::endl;
+
+using namespace caf;
+
+// Sends a random number of printable characters to buddy and exits
+void generator(event_based_actor* self, actor buddy) {
+  std::random_device rd;
+  std::minstd_rand gen{rd()};
+  const auto count = std::uniform_int_distribution<>{20, 100}(gen);
+  std::uniform_int_distribution<> dis{33, 126};
+  for (auto i = 0; i < count; i++) {
+    self->send(buddy, static_cast<char>(dis(gen)));
+  }
+}
+
+// Collects the incoming characters until either the awaited_size of characters
+// is received, or no new characters arrive for 100ms
+behavior collector(stateful_actor<std::string>* self, size_t awaited_size) {
```

Comment:
Not a fan of this `awaited_size`. I think it gets into the way of toying with this example by adding more generators for example.

My suggestion to print after a certain threshold was more about making sure that the buffer isn't growing indefinitely. So it would be more of a "flush threshold", but for an example I think we don't need this as a spawn parameter.

Reply:
Fair enough.